src/world/world.rs

```diff
@@ -146,23 +147,20 @@ impl<N: Scalar> World<N> {
     pub fn add_body(&mut self, rb: RigidBody<N>) -> RigidBodyHandle<N> {
         let position = rb.position().clone();
         let shape = rb.shape().clone();
-
-        let mut groups = CollisionGroups::new();
-        const STATIC_GROUP_ID: usize = 29;
-        groups.modify_whitelist(STATIC_GROUP_ID, false); // The static group is special.
+        let mut groups = rb.collision_groups().get_internal_collision_groups().clone();
 
         if rb.can_move() {
             // This is a dynamic object, remove it from the static objects group.
             groups.modify_membership(STATIC_GROUP_ID, false);
         }
         else {
-            // This is a dynamic object, keep is on the static objects group but prevent it from
-            // colliding with other static objects.
+            // This is a static object, keep it on the static objects group
+            // but prevent it from colliding with other static objects.
             groups.modify_blacklist(STATIC_GROUP_ID, true);
         }
```

Comment:
All those group manipulations should be removed now as this should have been done already when the rigid body was created.